src/main/java/se/iths/springbootgroupproject/configurations/SecurityConfig.java

```diff
@@ -0,0 +1,54 @@
+package se.iths.springbootgroupproject.configurations;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.access.hierarchicalroles.RoleHierarchy;
+import org.springframework.security.access.hierarchicalroles.RoleHierarchyImpl;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+import org.springframework.security.web.authentication.SimpleUrlAuthenticationSuccessHandler;
+import org.springframework.web.client.RestClient;
+import se.iths.springbootgroupproject.services.github.GithubOAuth2UserService;
+
+@Configuration
+public class SecurityConfig {
+
+    @Autowired
+    private GithubOAuth2UserService githubOAuth2UserService;
+
+    @Bean
+    SecurityFilterChain web(HttpSecurity http) throws Exception {
+        http
+                .authorizeHttpRequests((requests) -> requests
+                        .requestMatchers("/web/welcome", "/login", "/oauth/**", "/logout", "/error**").permitAll()
+                        .requestMatchers("/web/users/profiles/demo").authenticated()
+                        .anyRequest().authenticated()
```

Comment:
Maybe changing `.authenticated()` to `.denyAll()` for good security practice?

![image](https://github.com/Win-ther/SpringbootGroupProject/assets/143100902/5fcec836-76dc-4b39-a9dd-bc68e9596aae)